src/lib/operators/aggregate_hash.cpp

```diff
@@ -66,22 +70,28 @@ typename Results::reference get_or_add_result(CacheResultIds, ResultIds& result_
     // result in another 8 bytes being used.
     static_assert(std::is_same_v<AggregateKeyEntry, uint64_t>,
                   "Expected AggregateKeyEntry to be unsigned 64-bit value");
-    constexpr auto MASK = AggregateKeyEntry{1} << 63u;
 
     // Check if the AggregateKey already contains a stored index.
     if constexpr (std::is_same_v<CacheResultIds, std::true_type>) {
-      if (*first_key_entry & MASK) {
+      if (*first_key_entry & CACHE_MASK) {
         // The most significant bit is a 1, remove it by XORing the mask gives us the index into the results vector.
-        const auto result_id = *first_key_entry ^ MASK;
+        const auto result_id = *first_key_entry ^ CACHE_MASK;
 
         // If we have not seen this index as part of the current aggregate function, the results vector may not yet have
```

Comment:
Not seen is not checked here right? Just because we find a key that has not been seen before (which we do not track without looking at the ROW ID, does not mean anything for the following resizing.